devito/passes/iet/mpi.py

```diff
@@ -108,6 +108,27 @@ def _merge_halospots(iet):
     Merge HaloSpots on the same Iteration tree level where all data dependencies
     would be honored.
     """
+
+    # Merge rules -- if the retval is True, then it means the input `dep` is not
+    # a stopper to halo merging
+
+    def rule0(dep, hs, *args):
+        # E.g., `dep=W<f,[t1, x]> -> R<f,[t0, x-1]>` => True
+        return not any(d in hs.dimensions or dep.distance_mapper[d] is S.Infinity
+                       for d in dep.cause)
+
+    def rule1(dep, *args):
```

Comment:
Can wee keep same signature for all three `(dep,  hs, loc_indices)` for safety since they are all called the same way (kinda lie a template if one want to add more rules).

Reply:
OK, pushed fix